### PR TITLE
Sc 30945/client dmi partners inc pulled an export

### PR DESCRIPTION
**Pull request by richardfordjunior**
## Description of the change

> This fixes the issue where rich-types are not being properly masked in exports

## Type of change
- [X] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change
- [ ] Technical Debt
- [ ] Documentation

## Related tickets

> https://app.shortcut.com/active-prospect/story/30945/client-dmi-partners-inc-pulled-an-export-of-tf-data-service-info-for-leads-even-though-the-certs-are-not-masked-many-of-the

### Development and Testing

- [X]  Lint rules pass locally.
- [X]  The code changed/added as part of this pull request has been covered with tests, or this PR does not alter production code.
- [X]  All tests related to the changed code pass in development, or tests are not applicable.

### Code Review

- [X]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [X]  At least two engineers have been added as "Reviewers" on the pull request.
- [ ]  Changes have been reviewed by at least two other engineers who did not write the code.
- [X]  This branch has been rebased off master to be current.

### Tracking 
- [X]  Issue from Clubhouse has a link to this pull request.
- [X]  This PR has a link to the issue in Clubhouse.

### QA
- [ ]  This branch has been deployed to staging and tested.

### CODE REVIEW

**Review by asedarski — DISMISSED** [review]

File: lib/index.js [hunk 1]
```diff
@@ -55,9 +55,15 @@ const mask = function (obj, doMask = false) {
       for (const key in obj) {
         const value = obj[key];
         if (key === 'valid') { continue; }
-        obj[key] = mask(value, (obj.masked != null) || doMask);
+        obj[key] = mask(value, (obj.masked != null || obj.masked !== undefined) || doMask);
```
Comment (mikebetts) [inline lib/index.js:58]: I'm confused by this - if `obj.masked` is an object, won't it always return true and therefore mask despite what the masked value is set to in that object?

**Review by mikebetts — CHANGES_REQUESTED** [review]
Looks like this covers masking rich types in general, but it seems like this still isn't handling the case where the masked property itself is a rich type. It is supposed to look at its boolean value to determine whether to mask or not.

File: lib/index.js (hunk 1 above)
Comment (richardfordjunior) [reply lib/index.js:58]: I see what you mean. I'll update to cover when the `obj.masked` is a rich type.

Comment (richardfordjunior) [reply lib/index.js:58]: Yes.

**Review by asedarski — APPROVED** [review]
Provided the assertions in the tests are correct (I don't understand the business logic behind what's masked and what isn't), LGTM

**Pull request closed without merging**